app/uk/gov/hmrc/hec/controllers/SDESCallbackController.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.controllers
+
+import cats.data.EitherT
+import cats.implicits.catsSyntaxEq
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents}
+import uk.gov.hmrc.hec.models
+import uk.gov.hmrc.hec.models.sdes.CallBackNotification
+import uk.gov.hmrc.hec.models.sdes.NotificationStatus._
+import uk.gov.hmrc.hec.services.scheduleService.HECTaxCheckExtractionContext
+import uk.gov.hmrc.hec.services.{FileStoreService, TaxCheckService}
+import uk.gov.hmrc.hec.util.{FileMapOps, Logging}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.Future
+
+@Singleton
+class SDESCallbackController @Inject() (
+  fileStoreService: FileStoreService,
+  taxCheckService: TaxCheckService,
+  cc: ControllerComponents
+)(implicit hecTaxCheckExtractionContext: HECTaxCheckExtractionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  val taxCheckDir = "sdes/tax-checks"
+
+  val callback: Action[JsValue] = Action(parse.json).async { implicit request =>
+    Json.fromJson[CallBackNotification](request.body) match {
+      case JsSuccess(callbackNotification, _) =>
+        logger.info(
+          s"Received SDES callback for file: ${callbackNotification.filename}, " +
+            s"with correlationId : ${callbackNotification.correlationID} and status : ${callbackNotification.notification}"
+        )
+        callbackNotification.notification match {
+          case FileReady | FileReceived => Future.successful(Ok)
```

Comment:
what's the difference between `Ready` and `Received`?

Reply:
These are the status which come in the callback from SDES. i think FileReady means , file is Ready for pickup and FileReceived means file has been received by SDES